apps/experiments/models.py

```diff
@@ -396,16 +396,45 @@ class Experiment(BaseTeamModel):
     )
     use_processor_bot_voice = models.BooleanField(default=False)
 
+    # Versioning fields
+    working_experiment = models.ForeignKey(
+        "self",
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+        related_name="experiment_versions",
+    )
+    version_number = models.PositiveIntegerField(default=1)
+    is_default_version = models.BooleanField(default=False)
+    is_archived = models.BooleanField(default=False)
+
     class Meta:
         ordering = ["name"]
         permissions = [
             ("invite_participants", "Invite experiment participants"),
             ("download_chats", "Download experiment chats"),
         ]
+        constraints = [
+            models.UniqueConstraint(
+                fields=["is_default_version", "working_experiment"],
+                condition=Q(is_default_version=True),
+                name="unique_default_version_per_experiment",
+            ),
+            models.UniqueConstraint(
+                fields=["version_number", "working_experiment"],
+                condition=Q(working_experiment__isnull=False),
+                name="unique_version_number_per_experiment",
+            ),
+        ]
 
     def __str__(self):
         return self.name
 
+    def save(self, *args, **kwargs):
```

Comment:
I tried using a `CheckConstraint` to do this, but haven't had any luck. Not 100% if it's possible to do this with a check constraint, but it feels like it should be. I should maybe just read the docs again to understand the syntax better.